### DISCUSSION TITLE
How does this differ from the runas command?

### DISCUSSION THREAD

**Discussion by jarrodCoombes** [microsoft/sudo#66] [Q&A] [answered]
As the title says, I am curious about how windows `sudo` will differ from the `runas` command and why we need a second command to elevate commands in the CLI on Windows.

**Comment by AvogatoWizardWhisker** [comment, marked as answer]
From my understanding, `runas` in Windows allows the execution of commands or programs as any user, including administrators, whereas `sudo` permits the execution of elevated commands from a non-elevated console session. `runas` requires the password of the target user, offering the flexibility to operate as any user. In contrast, `sudo` only prompts for User Account Control (UAC) and simplifies the process by not requiring a separate password. Regarding permissions, `runas` grants broader permissions for the entire process, while `sudo` does not currently support running commands as other users, although this feature may be under consideration for future updates. In terms of security, `runas` necessitates careful password management. Conversely, `sudo` provides three configurations: `forceNewWindow`, which is deemed the most secure, along with `disableInput` and `normal`. Lastly, in terms of availability, `runas` is accessible in various Windows versions starting from Windows XP, while `sudo` is available from Windows 11 build 26052 onwards.

Here's documentation for `runas`: https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-R2-and-2012/cc771525(v=ws.11)

And for `sudo`: https://learn.microsoft.com/en-us/windows/sudo/

**Comment by jarrodCoombes** [comment]
Ahh, ok. So `runas` is more like `su` and as such you have to specify the other user (and password) where is `sudo` would allow elevation as the logged in user (assuming they have the needed permissions) without needing to enter a password. 

I think understand it now. Thanks @AvogatoWizardWhisker
